core/dynamo.c

```diff
@@ -383,7 +383,7 @@ get_dr_stats(void)
  * returns zero on success, non-zero on failure
  */
 DYNAMORIO_EXPORT int
-dynamorio_app_init(void)
+dynamorio_app_init(bool attach_case)
```

Comment:
I think we need some docs about how to set this parameter correctly.

Are we OK with changing the public API like this? We probably at least want to make a change notice to the release doc.

Reply:
It wouldn't be ok. But wait - this is not DR_API, this is DYNAMORIO_EXPORT. This shouldn't be public, is it?

Reply:
If this just indicates whether we're attaching, we already have dr_api_entry.

Reply:
Ok! Then consider this reverted. It really just needs to know whether this is the attach case, right.

Reply:
Oh yes, sorry! I confused tihs with "dr_app_setup" (which I was confused about, and thought was "dr_app_init").

Reply:
@Carrotman42 , still a good call. I didn't like this flag at all.

Reply:
Don't you really need to know whether *any* app code has already executed?  If so, that happens with more than just attaching.  Non-earliest-inject will go and run system code.  It's possible that code will run AVX-512 instructions.

Reply:
- see offline.

Reply:
Replaced with earliest inject flags, done.